modules/openapi-generator/src/main/resources/go-server/model.mustache

```diff
@@ -17,7 +17,39 @@ const (
 	{{#enumClassPrefix}}{{{classname.toUpperCase}}}_{{/enumClassPrefix}}{{name}} {{{classname}}} = {{{value}}}
 	{{/enumVars}}
 	{{/allowableValues}}
-){{/isEnum}}{{^isEnum}}{{#description}}
+)
+
+// All allowed values of {{{classname}}} enum
+var Allowed{{{classname}}}EnumValues = []{{{classname}}}{
+	{{#allowableValues}}
+	{{#enumVars}}
+	{{{value}}},
+	{{/enumVars}}
+	{{/allowableValues}}
+}
+
+// IsValid return true if the value is valid for the enum, false otherwise
+func (v {{{classname}}}) IsValid() bool {
+	for _, existing := range Allowed{{{classname}}}EnumValues {
+		if existing == v {
+			return true
+		}
+	}
+	return false
+}
+
+// New{{{classname}}}FromValue returns a pointer to a valid {{{classname}}}
+// for the value passed as argument, or an error if the value passed is not allowed by the enum
+func New{{{classname}}}FromValue(v {{{format}}}{{^format}}{{dataType}}{{/format}}) (*{{{classname}}}, error) {
```

Comment:
Could you help me understand why does this function need to return a pointer?

Reply:
This is the same implementation from the Go-Client Enum mustache template. By using the same implementation, a user can decide to use either the Client's models or the Server's models with only a couple minor problems to jump over.

If we want to change the signatures/implementation of the Go-Server Enum, I would also like to make the same changes to the Go-Client Enum Mustache Template.

Reply:
You don't need to modify the client as this doesn't affect the interaction between the client and server.

The changes only affect the server code internally. In this case, 

1. you are always dereferencing the return value in the controller
1. the pointer is only returned during an err, which means that the pointer will ultimately be discarded

It only makes sense to change the return type to a value.

Reply:
I made the change to not return a pointer from `New{{classname}}FromValue`.